pkg/controllers/securitypolicy_associated_controller.go

```diff
@@ -0,0 +1,250 @@
+			for _, port := range container.Ports {
+				if port.Name != "" {
+					podPortNames.Insert(port.Name)
+				}
+			}
+		}
+	}
+	return podPortNames
+}
+
+func reconcileSecurityPolicy(client client.Client, pods []v1.Pod, q workqueue.RateLimitingInterface) error {
+	podPortNames := getAllPodPortNames(pods)
+	log.V(1).Info("pod named port", "podPortNames", podPortNames)
+	spList := &v1alpha1.SecurityPolicyList{}
+	err := client.List(context.Background(), spList)
+	if err != nil {
+		log.Error(err, "failed to list security policy")
+		return err
+	}
+
+	for _, securityPolicy := range spList.Items {
+		shouldReconcile := false
+		for _, rule := range securityPolicy.Spec.Rules {
+			for _, port := range rule.Ports {
+				if port.Port.Type == intstr.String {
+					if podPortNames.Has(port.Port.StrVal) {
+						shouldReconcile = true
+					}
+				}
+			}
+		}
+		if shouldReconcile {
+			log.Info("reconcile security policy", "namespace", securityPolicy.Namespace, "name", securityPolicy.Name)
+			q.Add(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      securityPolicy.Name,
+					Namespace: securityPolicy.Namespace,
+				},
+			})
+		}
+	}
+	return nil
+}
+
+var PredicateFuncsPod = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		if p, ok := e.Object.(*v1.Pod); ok {
+			return checkPodHasNamedPort(*p, "create")
+		}
+		return false
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		if p, ok := e.ObjectOld.(*v1.Pod); ok {
+			if checkPodHasNamedPort(*p, "update") {
+				return true
+			}
+		}
+		if p, ok := e.ObjectNew.(*v1.Pod); ok {
+			if checkPodHasNamedPort(*p, "update") {
+				return true
+			}
+		}
+		return false
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		if p, ok := e.Object.(*v1.Pod); ok {
+			return checkPodHasNamedPort(*p, "delete")
+		}
+		return false
+	},
+}
+
+// checkPodHasNamedPort checks if the pod has a named port, it filters the pod events
+// we don't want give concern to.
+func checkPodHasNamedPort(pod v1.Pod, reason string) bool {
+	for _, container := range pod.Spec.Containers {
+		for _, port := range container.Ports {
+			if port.Name != "" {
+				log.Info("receive pod event", "reason", reason, "namespace", pod.Namespace, "name", pod.Name, "port",
+					port.Name, "portNumber", port.ContainerPort, "protocol", port.Protocol)
+				return true
+			}
+		}
+	}
+	return false
+}
+
+var PredicateFuncsNs = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return false
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldObj := e.ObjectOld.(*v1.Namespace)
+		newObj := e.ObjectNew.(*v1.Namespace)
+		log.Info("receive namespace event", "name", oldObj.Name)
+		if reflect.DeepEqual(oldObj.ObjectMeta.Labels, newObj.ObjectMeta.Labels) {
+			log.Info("label of namespace is not changed, ignore it", "name", oldObj.Name)
+			return false
+		}
+		if isSysNs, ok := oldObj.Annotations[WCP_SYSTEM_RESOURCE]; ok && strings.ToLower(isSysNs) == "true" {
```

Comment:
Same here, use a single function to check if it's system NS

Reply:
ok